### DISCUSSION TITLE
[RFC] good-first-issue / newbie label

### DISCUSSION THREAD

**Discussion by inclyc** [xmake-io/xmake#2908] [Ideas]
### Is your feature request related to a problem? Please describe.

The development of an open source community requires the participation of a large number of developers, and a project itself has accumulated a large amount of source code. It is not such trivival to get started and complete a large functional modification (and submit patch / GH PR). This RFC recommends adding `good first issue` to the github issues list to help those contributing to the `xmake` repository for the first time with some reasonably small changes. Similar to only need to change one or two lines in the source code, but does not affect the function can be regarded as `good first issue`

一个开源社区的发展需要大量开发者参与，而一个项目本身积累了大量的源代码，入门并完成一个较大的功能修改并不是一件简单的事情。这个RFC建议在github issues列表中加入`good first issue`来帮助那些第一次贡献给`xmake`仓库的人一些合理的小改动。类似于只需要在源代码中改一两行，但不影响功能的可以作为"good first issue"

### Describe the solution you'd like

Add `good first issue` label in https://github.com/xmake-io/xmake/labels shall be fine.

### Additional context

CC @waruqi

**Comment by waruqi** [comment]
label 这种是事后的，只有提了 pr 才会被标记，但是都提了 pr 了，有没有这个都无所谓了。。

如果只是想提一两行，随时可以提。

**Reply by inclyc** [reply]
大意是作为维护者可以留下一些简单的修改，让新人快速熟悉Codebase 例如 https://github.com/llvm/llvm-project/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22 

比如某个地方留下了一个小功能没有完全实现，但要实现其实非常简单，可以留着给社区实现？或许是一个不错的尝试。

**Reply by inclyc** [reply]
> label 这种是事后的，只有提了 pr 才会被标记，

可能误会我的意思了，是指issue的label而不是PR的label。可以写一些仓库目前存在的问题（简单的）然后依靠社区的力量开发

**Comment by waruqi** [comment]
这种老早有了，`contribution welcome` 和 `help wanted` ，很多 issues 我都标记了，就是希望有人贡献进来，没啥用，从来没人理过，到最后还是我自己搞掉。

https://github.com/xmake-io/xmake/issues?q=is%3Aissue+is%3Aopen+label%3A%22contribution+welcome%22

**Comment by inclyc** [comment]
> 很多 issues 我都标记了，就是希望有人贡献进来，没啥用，从来没人理过

https://github.com/xmake-io/xmake/blob/master/CONTRIBUTING.md

或许可以把链接加入到这个文档里？甚至README里？这样至少知道有这些label(

**Reply by waruqi** [reply]
> > 很多 issues 我都标记了，就是希望有人贡献进来，没啥用，从来没人理过
> 
> https://github.com/xmake-io/xmake/blob/master/CONTRIBUTING.md
> 
> 或许可以把链接加入到这个文档里？甚至README里？这样至少知道有这些label(

没用，CONTRIBUTING.md 和 readme 里面加了，也压根没人看。。

想贡献的，不提醒也会贡献，不想贡献的，到处加 link 也不会搭理。。

通常小改动 pr，都是为了解决用户自己遇到的问题，如果确实是小问题，我通常都会在 issues 里面直接回复，让他来个 pr 过来，告诉它修哪里。。也有不少人很乐意提 pr 过来，没必要这种 label.

如果不是自己遇到的 issues ，很少有人会主动来 pr 的

**Reply by inclyc** [reply]
Example: https://github.com/llvm/llvm-project/issues/56159

**Reply by inclyc** [reply]
> 如果不是自己遇到的 issues ，很少有人会主动来 pr 的

良好的社区还是应该有这种需求？总不可能一直让领导人一个人维护吧（？

**Comment by inclyc** [comment]
而且感觉已有的`contribution welcome`可能有些并不是很简单的修改就能完成的，新增这个label的目的是放一些简单的、需要熟悉代码仓库的issue作为 "**我为xmake解决的第一个issue**"

**Reply by waruqi** [reply]
> 而且感觉已有的`contribution welcome`可能有些并不是很简单的修改就能完成的，新增这个label的目的是放一些简单的、需要熟悉代码仓库的issue作为 "**我为xmake解决的第一个issue**"

真的如果是很简单的改动，我都懒得一直放着 issues 等人不知道什么时候过来提 pr，我分分钟就直接改了完事了。。

**Reply by inclyc** [reply]
> 我分分钟就直接改了完事了。。

某种意义上留给社区改可以促进社区的发展（

**Reply by inclyc** [reply]
> 某种意义上留给社区改可以促进社区的发展（

吸纳新的开发者的一个很好的来源

**Reply by waruqi** [reply]
放着不改，看着难受哈。。我还是直接改了吧。
